### PR TITLE
Modified search results page for elastic filtering and faceting. #6543

### PR DESCRIPTION
**Pull request by PritishC**
* Added a new macro for ES pagination, which is called inplace of regular pagination macro if facets are available.
* Added display_count fields via template filter.
* Removed `limit` arg from `auto_complete` classmethod.
* Updated copyright dates.

### CODE REVIEW

**Comment by tarunbhardwaj** [comment]
LGTM

**Comment by PritishC** [comment]
This PR needs to be refactored according to discussed refactor plan.

Working on it.

File: templates/webshop/search-results.jinja [hunk 1]
```diff
@@ -8,8 +8,37 @@
 {% endblock breadcrumb %}
 
 {% block main %}
+
 <section>
   <div class="container">
+  {% if facets %}
+  {% set facets = facets | add_display_counts %}
```
Comment (tarunbhardwaj) [inline templates/webshop/search-results.jinja]: Do not handle display_count logic for now. just display count for all

File: templates/webshop/search-results.jinja [hunk 2]
```diff
@@ -8,8 +8,37 @@
...
+      <form id="filter-form-sidebar">
+        <input type="hidden" name="q" value="{{ request.args.get('q') }}" />
+        <input type="hidden" name="page" value="1"/>
+        {% for key, value in facets.iteritems() %}
+        <ul>
+        <lh>{{ key }}</lh>
+        {% set facet_checked = request.args.getlist(key) %}
+        {% for term in value['terms'] %}
+          <li>
+            <input type="checkbox" name="{{ key }}" value="{{ term['term'] }}"
```
Comment (tarunbhardwaj) [inline templates/webshop/search-results.jinja]: I think you should move this logic to a macro with a block inside for inheritance.

```jinja
macro {
  macro_block {
    ...
  }
}
```

File: templates/webshop/_helpers.jinja [hunk 3]
```diff
@@ -142,6 +142,53 @@
 </ul>
 {% endmacro %}
 
+{% macro render_es_pagination(pagination, uri, endpoint) %}
```
Comment (tarunbhardwaj) [inline templates/webshop/_helpers.jinja]: no use of uri

**Pull request merged**